### DISCUSSION TITLE
Declare packages as installed on host

### DISCUSSION THREAD

**Discussion by L0ric0** [microsoft/vcpkg#21828] [Ideas]
****

**Is your feature request related to a problem? Please describe.**
When building programms for a complex infrastrukture like a cluster it would be nice do be able to define certain packages as already installed on the machine so that one dosen't has to build tings like openmpi and figure out the right configuration for the environment.

**Proposed solution**
As this configuration sould pobably not go into the project directory one could add a config file for vcpkg to `~/.config/vcpkg` to define this packages. For the defined packages it is asumed that they are installed to system paths (or the path are in the aplicable environment variables) and can be fond by find_package.

**Describe alternatives you've considered**
 * Modifing the vcpkg.json manifest for the project on the cluster, but this dosn't work dependencies of packages that are not installed.
 * Adding the config file to the project root, but that would then not apply to other projects build on the system and would explictly be removed from source control

**Additional context**
Building certain common packages on clusters (like `boost`, `hdf5`, `openmpi`) takes a long time and normaly a recent enough version is already configured and installed on the system.

**Comment by PhoebeHui** [comment]
@L0ric0, I'm not sure if I understand the case correctly, currently we have .\installed\vcpkg\status file to record the installed package, and 'vcpkg list' would display the installed package as well.

**Comment by L0ric0** [comment]
It is not to get the installed packages, but to tell vcpkg that a certain package is installed and doesn't need to be build.

So lets say i have `openmpi` installed from a distro package and a Project that depands on boost and i know that the already installed `openmpi` meets the requirements of the project, but have another host on which the project also has to run where `openmpi` is not installed. so i define a `vcpkg.json` file that lists `openmpi` in its dependencies. At the moment both on the host that has `openmpi` already installed and the host where it is not openmpi is build. What i propose is a file where the vcpkg user can define which packages do not need to be build as the host system already provides them. So the host with `openmpi` already installed would not have to build it again and the user wouldn't have to worry about its configuration for this particular environment.

**Comment by PhoebeHui** [comment]
I see, could you check if the [binary caching](https://github.com/microsoft/vcpkg/blob/master/docs/specifications/binarycaching.md) meet the requirements?

**Comment by NancyLi1013** [comment]
Hi @L0ric0 Have you tried the above suggestion form @PhoebeHui? Does it work for you?

**Comment by JackBoosY** [comment]
I strongly doubt the need to add this feature.
The pre-compiled binary downloaded manually by the user has a different version/different configuration from the vcpkg in most cases, so other errors may occur.
